web/pdf_viewer.js

```diff
@@ -646,23 +643,24 @@ class PDFViewer {
 
     // Handle the window/tab becoming inactive *after* rendering has started;
     // fixes (another part of) bug 1746213.
-    const visibilityChangePromise = new Promise(resolve => {
-      this.#onVisibilityChange = () => {
-        if (document.visibilityState !== "hidden") {
-          return;
-        }
-        resolve();
-      };
-      document.addEventListener("visibilitychange", this.#onVisibilityChange);
+    const hiddenCapability = Promise.withResolvers();
+    function onVisibilityChange() {
+      if (document.visibilityState === "hidden") {
+        hiddenCapability.resolve();
+      }
+    }
+    document.addEventListener("visibilitychange", onVisibilityChange, {
+      signal,
     });
 
     await Promise.race([
       this._onePageRenderedCapability.promise,
-      visibilityChangePromise,
+      hiddenCapability.promise,
     ]);
-    // Ensure that the "visibilitychange" listener is always removed.
-    document.removeEventListener("visibilitychange", this.#onVisibilityChange);
-    this.#onVisibilityChange = null;
+    // Ensure that the "visibilitychange" listener is removed immediately.
+    document.removeEventListener("visibilitychange", onVisibilityChange, {
+      signal,
+    });
```

Comment:
`removeEventListener` does not accept a signal.
```suggestion
    // Ensure that the "visibilitychange" listener is removed immediately.
    document.removeEventListener("visibilitychange", onVisibilityChange);
```